docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -158,7 +160,23 @@ Grantee: cosmos1ghi...jkl
 End of transaction messages
 ```
 
-### 9. Signing Payload and Wire Format
+### 10. Require signing over the `TxBody` and `AuthInfo` raw bytes
+
+Recall that the transaction bytes merklelized on chain are the Protobuf binary serialization of [TxRaw](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/proto/cosmos/tx/v1beta1/tx.proto#L33), which contains the `body_bytes` and `auth_info_bytes`. Moreover, the transaction hash is defined as the SHA256 hash of the `TxRaw` bytes. We require that the user signs over these canonical bytes in SIGN_MODE_TEXTUAL, more specifically over the following string:
+
+```
+*Hash of raw bytes: <base64(sha256(<body_bytes> ++ " " ++ <auth_info_bytes>))>
+```
+
+where `++` denotes concatenation.
+
+This is to prevent transaction hash malleability. The point #1 about bijectivity assures that transaction `body` and `auth_info` values are not malleable, but the transaction hash still might be malleable with point #1 only, because the SIGN_MODE_TEXTUAL strings don't follow the byte ordering defined in `body_bytes` and `auth_info_bytes`. Without this hash, a malicious validator or exchange could intercept a transaction, modify its transaction hash _after_ the user signed it using SIGN_MODE_TEXTUAL (by tweaking the byte ordering inside `body_bytes` or `auth_info_bytes`), and then submit it to Tendermint.
```

Comment:
> SIGN_MODE_TEXTUAL strings don't follow the byte ordering defined in `body_bytes` and `auth_info_bytes`

I looked but couldn't find anything -- is there a link or doc or something that provides detail about this? The on-chain representation of data is authoritative, and byte ordering of on-chain data is well-defined, so this is confusing.

Reply:
Everything is in the ADR. Textual is a `Proto Tx <-> []string` encoding, not `Proto TxRaw <-> []string`.

Reply:
> Textual is `Proto Tx <-> []string` encoding, not `Proto TxRaw <-> []string`.

I hope my comments have made it clear why this isn't a valid way to produce signatures 🤷

Reply:
Your thesis is clear. However, I do not agree. Again, let's discuss this elsewhere since it's touches signing more generally than this PR 👍.

Reply:
I agree with Amaury. Simple proof is an observation that bijectivity is transitive (not sure if this is the proper way to say that). In other words: if `Textual: Tx->string` and `Proto: Tx -> bytes` are bijective, then we can have `Textual . Proto': bytes -> string` which is also bijective.